drivers/amazonec2/amazonec2_test.go

```diff
@@ -101,7 +102,9 @@ func TestConfigureSecurityGroupPermissionsDockerAndSsh(t *testing.T) {
 func TestConfigureSecurityGroupPermissionsSkipReadOnly(t *testing.T) {
 	driver := NewTestDriver()
 	driver.SecurityGroupReadOnly = true
-	perms, err := driver.configureSecurityGroupPermissions(securityGroup)
+	group := securityGroup
+	group.IpPermissions = []*ec2.IpPermission{}
```

Comment:
can we define another securityGroup like [this](https://github.com/rancher/machine/blob/91e16c70958c58ee10980db0d9e0606f1e01ae8e/drivers/amazonec2/amazonec2_test.go#L27), but with IpPermissions and use that for this test, and also in `TestConfigureSecurityGroupPermissionsOpenPorts` and `TestConfigureSecurityGroupPermissionsInvalidOpenPorts` tests

Reply:
All the tests currently use the securityGroup you linked above; however, I reset the group.IpPermissions here since it seems like using the same test resource was causing the go test to leak the IpPermissions from one test to another (tested this by running tests individually, where they all pass, and then together, where some fail due to unexpected IpPermissions).

As discussed offline, I defined two test resources now instead! Will push them in the latest commit.